src/libtriton/format/pe/pe.cpp

```diff
@@ -0,0 +1,204 @@
+      }
+
+
+      triton::uint64 PE::getOffsetFromAddress(triton::uint64 vaddr) const {
+        for (auto section : this->header.getSectionHeaders()) {
+          if (vaddr >= section.virtualAddress && vaddr < (section.virtualAddress + section.rawSize)) {
+              return ((vaddr - section.virtualAddress) + section.rawAddress);
+          }
+        }
+        return 0;
+      }
+
+      void PE::initExportTable(void) {
+          triton::uint32 exportStart = header.getDataDirectory().exportTable_rva;
+          triton::uint32 exportSize = header.getDataDirectory().exportTable_size;
+          if (0 == exportStart) {
+              std::memset(&exportTable, '\0', 40);
+              return;
+          }
+          std::memcpy(&exportTable, raw+getOffsetFromAddress(exportStart), 40);
+          exportTable.name = std::string((char*)raw+getOffsetFromAddress(exportTable.nameRVA));
+          triton::uint32 addrTableStart = getOffsetFromAddress(exportTable.exportAddressTableRVA);
+          for (triton::usize i=0; i<exportTable.addressTableEntries; ++i) {
+              PEExportEntry entry;
+              triton::uint32 exportRVA;
+              std::memcpy(&exportRVA, raw+addrTableStart+4*i,4);
+              if (exportRVA >= exportStart && exportRVA < exportStart+exportSize) {
+                  entry.isForward = true;
+                  entry.forwarderRVA = exportRVA;
+                  entry.forwarderName = std::string((char*)raw+getOffsetFromAddress(exportRVA));
+              } else {
+                  entry.isForward = false;
+                  entry.exportRVA = exportRVA;
+              }
+              exportTable.entries.push_back(entry);
+          }
+          triton::uint32 nameTableStart = getOffsetFromAddress(exportTable.namePointerRVA);
+          triton::uint32 ordTableStart = getOffsetFromAddress(exportTable.ordinalTableRVA);
+          for (triton::usize i=0;i<exportTable.numberOfNamePointers; ++i) {
+              triton::uint16 ordinal;
+              std::memcpy(&ordinal, raw+ordTableStart+2*i,2);
+              triton::uint32 nameRVA;
+              std::memcpy(&nameRVA, raw+nameTableStart+4*i,4);
+              exportTable.entries[ordinal].ordinal = ordinal;
+              exportTable.entries[ordinal].exportNameRVA = nameRVA;
+              exportTable.entries[ordinal].exportName = std::string((char*)raw+getOffsetFromAddress(nameRVA));
+          }
+      }
+
+      void PE::initImportTable(void) {
+          triton::uint32 importStart = header.getDataDirectory().importTable_rva;
+          if (0 == importStart) return;
+          triton::uint32 importOffset = getOffsetFromAddress(importStart);
+          triton::uint64 byNameMask = (header.getOptionalHeader().magic == PE_FORMAT_PE32PLUS ?
+            0x8000000000000000 : 0x80000000);
+          triton::uint32 entrySize = (header.getOptionalHeader().magic == PE_FORMAT_PE32PLUS ? 8 : 4);
+          triton::uint32 pos = importOffset;
+          PEImportDirectory impdt;
```

Comment:
`PEImportDirectory` should be a class. Take a look how Elf parses its structures.